### ISSUE TITLE
bug: Contractions are incorrectly reported as typos

### ISSUE THREAD

**Issue by nfelt14** [pawamoy/mkdocs-spellcheck#22]
### Description of the bug

Basic contractions such as "Doesn't" or "couldn't" are being reported as typos because the checker is splitting the word on the single quote character.

### To Reproduce

1. Create a document with the word "doesn't" in it
2. Build the docs with the symspellpy backend

### Full traceback

```console
WARNING -  mkdocs_spellcheck: (symspellpy) index.md: Misspelled 'doesn', did you mean 'does'?
```

### Expected behavior

The contraction should not be flagged as a typo.

### Environment information

```console
$ python -m mkdocs_spellcheck.debug
- __System__: Windows-10-10.0.19045-SP0
- __Python__: cpython 3.11.8
- __Environment variables__:
- __Installed packages__:
  - `mkdocs-spellcheck` v1.0.3
```

**Comment by pawamoy** [comment]
Yep, I'm aware of this limitation. There's no easy solution though. Each language has its own peculiarities. We could rely on natural language processing libraries to correctly split text. Not sure how easy it would be :) Happy to see more suggestions and/or review PRs that address this!

**Comment by pawamoy** [comment]
Closing in favor of #3.
